### PR TITLE
Remove unused dependencies

### PR DESCRIPTION
**Pull request by atuchin-m**
Resolves https://github.com/brave/brave-browser/issues/18129

Discussed in:
https://bravesoftware.slack.com/archives/C7VLGSR55/p1631686319300400

Was initially added with `brave_browser_test` target in https://github.com/brave/brave-core/pull/65/files
Before merge we have to check that CI will still build `chrome` target.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by mariospr — APPROVED** [review]
LGTM!

**Review by iefremov — APPROVED** [review]

**Comment by atuchin-m** [comment]
In fact, compilation of chromium `browser_tests` had already being broken.
I've fixed them by the last commit.
For the `brave_broweser_tests` nothing has changed: the number and the correctness are the same.

File: test/base/testing_brave_browser_process.cc [hunk 1]
```diff
@@ -3,6 +3,8 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+ #include "brave/test/base/testing_brave_browser_process.h"
```
Comment (goodov) [inline test/base/testing_brave_browser_process.cc:6]: this include already exists at line 13, you should remove it then from line 13.

File: patches/chrome-test-BUILD.gn.patch [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
-     ]
+@@ -179,6 +179,8 @@ static_library("test_support") {
+     "base/tracing.h",
+   ]
  
-+    deps += [ "//brave/test:brave_browser_tests_deps", ]
-     # Runtime dependencies
-     data_deps = [
-       "//chrome:browser_tests_pak",
++  import("//brave/chromium_src/chrome/test/sources.gni") sources += brave_chromium_src_chrome_test_test_support_sources
```
Comment (goodov) [inline patches/chrome-test-BUILD.gn.patch:9]: should be done in one line: 
```import("//brave/chromium_src/chrome/test/sources.gni") sources += brave_chromium_src_chrome_test_test_support_sources deps += brave_chromium_src_chrome_test_test_support_deps```

File: chromium_src/chrome/test/sources.gni [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2021 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+brave_chromium_src_chrome_test_test_support_sources = [
+ "//brave/test/base/test_brave_browser_process.cc",
+ "//brave/test/base/test_brave_browser_process.h",
+]
```
Comment (goodov) [inline chromium_src/chrome/test/sources.gni:9]: it's better to add deps here and include them in the patch.
also add a newline at the end of the file. currently there is no newline.

File: test/base/testing_brave_browser_process.cc (hunk 1 above)
Comment (atuchin-m) [reply test/base/testing_brave_browser_process.cc:6]: Done

File: patches/chrome-test-BUILD.gn.patch (hunk 2 above)
Comment (atuchin-m) [reply patches/chrome-test-BUILD.gn.patch]: done

File: chromium_src/chrome/test/sources.gni (hunk 3 above)
Comment (atuchin-m) [reply chromium_src/chrome/test/sources.gni]: fixed

**Review by goodov — APPROVED** [review]

**Pull request merged**